app/models/banned_password.rb

```diff
@@ -0,0 +1,24 @@
+require "rest-client"
+
+class BannedPassword < ApplicationRecord
+  NCSC_DENYLIST_URL = "https://ncsc.gov.uk/static-assets/documents/PwnedPasswordsTop100k.txt".freeze
+
+  def self.import_from_ncsc
+    response = RestClient.get NCSC_DENYLIST_URL
+    _header, passwords = response.body.lines.map(&:rstrip).to_a.split("--")
```

Comment:
I'm more than likely doing something weird locally but I was getting an `undefined method 'split' for #<Array..` error with this row, I was successful with the following (including testing the lower case):
```suggestion
    _header, passwords = response.body.downcase!.split("--").map(&:split)
```

Reply:
Rails patches a lot of built in things (including arrays), so it can be hard to run some code in a normal ruby shell.  I've added a test stubbing the file and verifying that it's imported properly

Reply:
Huh, what an interesting concept. I figured something like that was perhaps happening, although I was running in the Rails console from the root of this repo 🤷‍♂️ but good to know it's just me not getting it working!